src/psyclone/psyir/nodes/node.py

```diff
@@ -1135,15 +1139,21 @@ def ancestor(self, my_type, excluding=None, include_self=False,
 
         # We search up the tree until we find an ancestor of the requested
         # type(s) shared by the shared_with node.
-        while (myparent is not shared_ancestor and myparent is not None and
-               shared_ancestor is not None):
+        while (myparent is not shared_ancestor and myparent and
+                shared_ancestor):
             if myparent is limit:
                 break
             if myparent.depth > shared_ancestor.depth:
+                # If myparent is deeper in the tree than the shared_ancestor
```

Comment:
Please could you re-phrase this? Perhaps s/the potential shared ancestor so/then/

Reply:
Changed the comments, should be better now.